src/module/helpers/registry.mjs

```diff
@@ -0,0 +1,123 @@
+/**
+ * @import { RegistryEntry } from "./_types";
+ */
+
+/**
+ * A registry of cached document info compiled across all compendiums.
+ * Each collection is expected to be of a a single document type & subtype.
+ */
+export default class DrawSteelRegistry {
+  /**
+   * Called once in `ready` after migrations.
+   */
+  async initialize() {
+    const itemPacks = game.packs.filter(p => p.documentName === "Item").sort((a, b) => {
+      // always work through system compendiums first
+      const typeMap = {
+        system: 0,
+        module: 1,
+        world: 2,
+      };
+
+      return typeMap[a.metadata.packageType] - typeMap[b.metadata.packageType];
+    });
+
+    for (const pack of itemPacks) {
+      // Fancy performant fetch
+      const docs = await pack.getDocuments({ type__in: ["class", "subclass", "perk", "kit"] });
+
+      for (const item of docs) {
+        const dsid = item.dsid;
+        /** @type {RegistryEntry} */
+        const registryEntry = {
+          dsid,
+          name: item.name,
+          uuid: item.uuid,
+        };
+        switch (item.type) {
+          case "class":
+            registryEntry.primary = item.system.primary;
+            if (this.class.has(item.dsid)) {
+              console.warn(`Replacing ${item.type} registry entry for ${dsid}`);
+            }
+            this.class.set(dsid, registryEntry);
+            break;
+          case "subclass":
+            registryEntry.classLink = item.system.classLink;
+            if (this.subclass.has(item.dsid)) {
+              console.warn(`Replacing ${item.type} registry entry for ${dsid}`);
+            }
+            this.subclass.set(dsid, registryEntry);
+            break;
+          case "perk":
+            registryEntry.perkType = item.system.perkType;
+            if (this.perk.has(item.dsid)) {
+              console.warn(`Replacing ${item.type} registry entry for ${dsid}`);
+            }
+            this.perk.set(dsid, registryEntry);
+            break;
+          case "kit":
+            if (this.kit.has(item.dsid)) {
+              console.warn(`Replacing ${item.type} registry entry for ${dsid}`);
+            }
+            this.kit.set(dsid, registryEntry);
+            break;
+        }
+      }
+    }
+  }
+
+  /* -------------------------------------------------- */
+  /*  Registries                                        */
+  /* -------------------------------------------------- */
```

Comment:
The private properties are missing annotations, and the getters are missing `@type`.